sdk/eventhub/azure-eventhub/azure/eventhub/_common.py

```diff
@@ -322,6 +381,82 @@ def body_as_json(self, encoding="UTF-8"):
         except Exception as e:
             raise TypeError("Event data is not compatible with JSON type: {}".format(e))
 
+    @property
+    def content_type(self):
+        # type: () -> Optional[str]
+        """The content type descriptor.
+        Optionally describes the payload of the message, with a descriptor following the format of RFC2045, Section 5,
+        for example "application/json".
+        :rtype: str
+        """
+        if not self._raw_amqp_message.properties:
+            return None
+        try:
+            return self._raw_amqp_message.properties.content_type.decode("UTF-8")
+        except (AttributeError, UnicodeDecodeError):
+            return self._raw_amqp_message.properties.content_type
+
+    @content_type.setter
+    def content_type(self, value):
+        # type: (str) -> None
+        if not self._raw_amqp_message.properties:
+            self._raw_amqp_message.properties = AmqpMessageProperties()
+        self._raw_amqp_message.properties.content_type = value
+
+    @property
+    def correlation_id(self):
+        # type: () -> Optional[str]
+        # pylint: disable=line-too-long
+        """The correlation identifier.
+        Allows an application to specify a context for the message for the purposes of correlation, for example
+        reflecting the MessageId of a message that is being replied to.
+        :rtype: str
+        """
+        if not self._raw_amqp_message.properties:
+            return None
+        try:
+            return self._raw_amqp_message.properties.correlation_id.decode("UTF-8")
+        except (AttributeError, UnicodeDecodeError):
+            return self._raw_amqp_message.properties.correlation_id
+
+    @correlation_id.setter
+    def correlation_id(self, value):
+        # type: (str) -> None
+        if not self._raw_amqp_message.properties:
+            self._raw_amqp_message.properties = AmqpMessageProperties()
+        self._raw_amqp_message.properties.correlation_id = value
+
+    @property
+    def message_id(self):
+        # type: () -> Optional[str]
+        """The id to identify the message.
+        The message identifier is an application-defined value that uniquely identifies the message and its payload.
+        The identifier is a free-form string and can reflect a GUID or an identifier derived from the
+        application context.  If enabled, the duplicate detection feature identifies and removes second and
+        further submissions of messages with the same message id.
+        :rtype: str
+        """
+        if not self._raw_amqp_message.properties:
+            return None
+        try:
+            return self._raw_amqp_message.properties.message_id.decode("UTF-8")
+        except (AttributeError, UnicodeDecodeError):
+            return self._raw_amqp_message.properties.message_id
+
+    @message_id.setter
+    def message_id(self, value):
+        # type: (str) -> None
+        if value and len(str(value)) > MESSAGE_PROPERTY_MAX_LENGTH:
```

Comment:
I guess the MESSAGE_PROPERTY_MAX_LENGTH is copied from the service bus, but I don't remember the eventhubs service has mentioned such restriction.

I would suggest testing setting message_id longer than MESSAGE_PROPERTY_MAX_LENGTH in event hubs, and drop the condition if it works with eh

Reply:
dropped